lib/python/picongpu/plugins/plot_mpl/energy_histogram_visualizer.py

```diff
@@ -0,0 +1,131 @@
+"""
+This file is part of the PIConGPU.
+
+Copyright 2017-2018 PIConGPU contributors
+Authors: Sebastian Starke
+License: GPLv3+
+"""
+
+from picongpu.plugins.energy_histogram import EnergyHistogram
+import matplotlib.pyplot as plt
+
+
+class Visualizer:
+    """
+    Class for creation of histogram plots on a logscaled y-axis.
+    """
+
+    def __init__(self, run_directory):
+        """
+        Parameters
+        ----------
+        run_directory : string
+            path to the run directory of PIConGPU
+            (the path before ``simOutput/``)
+        """
+        if run_directory is None:
+            raise ValueError('The run_directory parameter can not be None!')
+
+        self.energy_histogram = EnergyHistogram(run_directory)
+        self.plt_obj = None
+
+    def visualize(self, iteration, ax, species='e', species_filter="all",
+                  **kwargs):
+        """
+        Creates a semilogy plot on the provided axes object for
+        the data of the given iteration using matpotlib.
+
+        Parameters
+        ----------
+        iteration: int
+            the iteration number for which data will be plotted.
+        ax: matplotlib axes object
```

Comment:
a default for `ax` could be `plt.gca()`

Reply:
This default leads to the creation of another empty plot even when an ax argument is given, so I don't think we should do that...

Reply:
oh, ok.
I was just thinking you do something like:
```python
def func(ax=None, ...)
    if ax is None:
        ax = plt.gca()
```

Reply:
yeah, that's what came to my mind as well...I'll add that